### ISSUE TITLE
Add correct URL to s3 backend error message

### ISSUE THREAD

**Issue by mbtamuli** [hashicorp/terraform#13410]
### Terraform Version
```Terraform v0.9.2```

### Affected Resource(s)
- backend "s3" documentation

### Terraform Configuration Files
```hcl
terraform {
  backend "s3" {
    bucket = "tfstate-bucket"
    key    = "terraform.tfstate"
    region = "us-east-1"
    encrypt = "true"
  }
}
```

### Expected Behavior
It shouldn't link to a nonexistent page. https://www.terraform.io/docs/state/remote/s3.html

### Actual Behavior
On terraform apply, because my `~/.aws/credentials` file was not created and I had not specified the access key and the secret key, it showed an error.
```
terraform init
Initializing the backend...

Error configuring the backend "s3": 1 error(s) occurred:

* No valid credential sources found for AWS S3 remote.
Please see https://www.terraform.io/docs/state/remote/s3.html for more information on
providing credentials for the AWS S3 remote

Please update the configuration in your Terraform files to fix this error
then run this command again.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Confgure s3 backend.
2. Provide no credentials and don't create AWS credentials file(~/.aws/credentials).
3. `terraform init`

**Comment by shdobxr** [comment]
Same version of Terraform (windows) and indeed I am puzzled by the dead link as well.

**Comment by bflad** [comment]
The fix for the error messaging has been merged and will release with version 0.13.0-beta2 of Terraform.
